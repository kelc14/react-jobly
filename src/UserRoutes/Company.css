.Company-container {
  width: 70vw;
  background-color: white;
  border: 1px solid black;
  border-radius: 10px;
  margin: 0 auto;
  padding: 20px;
  overflow: hidden;
  text-align: left;
  margin-bottom: 25px;
}

.Company-company {
  text-align: left;
  font-weight: bold;
  font-size: 1rem;
}

.Company-about {
  text-align: left;
  font-size: 0.8rem;
}

.Company-logo {
  float: right;
}
