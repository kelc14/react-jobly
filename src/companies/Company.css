.Company-container {
  width: 70vw;
  background-color: white;
  border: 1px solid black;
  border-radius: 10px;
  margin: 0 auto;
  padding: 20px;
  overflow: hidden;
  text-align: left;
  margin-bottom: 25px;
}

.Company-container:hover {
  background-color: goldenrod;
}

.Company-link {
  text-decoration: none;
  color: black;
}

.Company-company {
  text-align: left;
  font-weight: bold;
  font-size: 1rem;
}

.Company-about {
  text-align: left;
  font-size: 0.8rem;
}

.Company-logo {
  float: right;
}
