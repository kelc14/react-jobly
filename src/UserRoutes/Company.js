import React from "react";
import "./Company.css";

const Company = () => {
  return (
    <div className="Company-container">
      <p className="Company-company"> Company Name Goes Here </p>
      <p className="Company-about"> Description: 1,0000000 </p>
      <p className="Company-about"> Number of Employees: 00000 </p>
      <img src={`img logo here`} alt="logo" className="Company-logo" />
    </div>
  );
};

export default Company;
