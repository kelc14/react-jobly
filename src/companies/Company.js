import React from "react";
import "./Company.css";
import { Link } from "react-router-dom";

const Company = ({ data }) => {
  const { name, handle, description, numEmployees, logoUrl } = data;
  return (
    <Link to={`./${handle}`} className="Company-link">
      <div className="Company-container">
        <p className="Company-company"> {name} </p>
        <p className="Company-about"> Description: {description} </p>
        <p className="Company-about"> Number of Employees: {numEmployees} </p>
        {logoUrl ? (
          <img src={`${logoUrl}`} alt="logo" className="Company-logo" />
        ) : (
          <></>
        )}
      </div>{" "}
    </Link>
  );
};

export default Company;
