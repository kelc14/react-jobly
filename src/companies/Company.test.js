import { render, act } from "@testing-library/react";
import Company from "./Company";
import JoblyApi from "../api/api";
import { MemoryRouter } from "react-router-dom";

let testCompany = {
  handle: "test-company",
  name: "Test Company",
  description: "This is a test company for mocking our API calls.",
  numEmployees: 1000,
  logoUrl: null,
};
/**
 * SMOKE TEST
 */
test("Company renders without crashing", () => {
  render(
    <MemoryRouter>
      {" "}
      <Company data={testCompany} />
    </MemoryRouter>
  );
});

/**
 * SNAPSHOT TEST
 */
test("Company matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      {" "}
      <Company data={testCompany} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
