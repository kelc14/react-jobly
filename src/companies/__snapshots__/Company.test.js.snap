// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company matches snapshot 1`] = `
<DocumentFragment>
   
  <a
    class="Company-link"
    href="/test-company"
  >
    <div
      class="Company-container"
    >
      <p
        class="Company-company"
      >
         Test Company 
      </p>
      <p
        class="Company-about"
      >
         Description: This is a test company for mocking our API calls. 
      </p>
      <p
        class="Company-about"
      >
         Number of Employees: 1000 
      </p>
    </div>
     
  </a>
</DocumentFragment>
`;
